src/admin/migrations/m170515_115236_basetables.php

```diff
@@ -22,29 +22,35 @@ public function safeUp()
         ]);
         
         $this->addPrimaryKey('estore_product_group_pk', '{{%estore_product_group_ref}}', ['group_id', 'product_id']);
-        
+
         $this->createTable('{{%estore_product}}', [
             'id' => $this->primaryKey(),
             'name' => $this->text()->notNull(), // Unicorm Trousers
-            'producer_id' => $this->integer()->notNull(),
+            'producer_id' => $this->integer()->notNull(), // @todo: ref to parent product?
+            'visibility' => $this->integer()->defaultValue(1)->notNull(), // 0 = not visible, 1 visible anywhere, 2 = only in category, 3 only in search
+            'enabled' => $this->boolean()->defaultValue(false)->notNull(),
         ]);
         
         $this->createTable('{{%estore_set}}', [
             'id' => $this->primaryKey(),
-            'name' => $this->string()->notNull(), // Trousers
+            'code' => $this->string()->unique()->notNull(), // trousers
+            'name' => $this->string()->unique()->notNull(), // Trousers
         ]);
-        
+
         $this->createTable('{{%estore_product_set_ref}}', [
             'product_id' => $this->integer()->notNull(),
             'set_id' => $this->integer()->notNull(),
         ]);
-        
+
         $this->addPrimaryKey('estore_product_set_ref_pk', '{{%estore_product_set_ref}}', ['product_id', 'set_id']);
-        
+        $this->addForeignKey('estore_product_set_ref_product_id_fk', '{{%estore_product_set_ref}}', ['product_id'], '{{%estore_product}}', ['id'], 'CASCADE', 'CASCADE');
+        $this->addForeignKey('estore_product_set_ref_set_id_fk', '{{%estore_product_set_ref}}', ['set_id'], '{{%estore_set}}', ['id'], 'CASCADE', 'CASCADE');
+
         $this->createTable('{{%estore_set_attribute}}', [
             'id' => $this->primaryKey(),
             'type' => $this->integer(), // 1 = integer, 2 = boolean, 3 = string
             'input' => $this->string()->notNull(), // zaa-text, zaa-password
+            'code' => $this->string()->notNull(), // size, color, material_type, width, height
```

Comment:
Its a slug? should be unique? For url generation? or maybe its "identifier"?

Reply:
yes right should be also unique.